### PR TITLE
fix: AlreadyDisposedException due to Project disposal before using project's service [ROAD-606]

### PR DESCRIPTION
**Pull request by ArtsiomCh**
For all our project services create correspondent funs in Util and check there if project is disposed return Null otherwise the service instance.
Then we need to remember to use any such service from correspondent fun only. 
Also Null case need to be handled in each particular case (in most places it looks like just ?. usage needed).

### CODE REVIEW

File: src/integTest/kotlin/io/snyk/plugin/services/SnykTaskQueueServiceHeavyTest.kt [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+import io.snyk.plugin.resetSettings
+import org.junit.Test
+
+@Suppress("FunctionName")
+class SnykTaskQueueServiceHeavyTest : HeavyPlatformTestCase() {
+
+    override fun setUp() {
+        super.setUp()
+        unmockkAll()
+        resetSettings(project)
```
Comment (bastiandoetsch) [inline src/integTest/kotlin/io/snyk/plugin/services/SnykTaskQueueServiceHeavyTest.kt:17]: Could you please mock Sentry here, so we don't get error reporting? Something like `mockkStatic(Sentry::class)`

File: src/main/kotlin/io/snyk/plugin/SnykBulkFileListener.kt [hunk 2]
```diff
@@ -114,7 +112,7 @@ class SnykBulkFileListener : BulkFileListener {
                 }
 
             if (filesToRemoveFromCache.isNotEmpty())
-                project.service<SnykTaskQueueService>().scheduleRunnable("Snyk Code is updating caches...") {
+                getSnykTaskQueueService(project)?.scheduleRunnable("Snyk Code is updating caches...") {
```
Comment (bastiandoetsch) [inline src/main/kotlin/io/snyk/plugin/SnykBulkFileListener.kt:115]: Please look at the Detekt warning - there seem to be braces missing...

File: src/main/kotlin/io/snyk/plugin/Utils.kt [hunk 3]
```diff
@@ -34,13 +38,25 @@ import javax.net.ssl.TrustManager
...
+
+fun getSnykToolWindowPanel(project: Project): SnykToolWindowPanel? = project.serviceIfNotDisposed()
+
+fun getAmplitudeExperimentService(project: Project): AmplitudeExperimentService? = project.serviceIfNotDisposed()
+
+fun getSnykCliAuthenticationService(project: Project): SnykCliAuthenticationService? = project.serviceIfNotDisposed()
+
+fun getSnykCliDownloaderService(project: Project): SnykCliDownloaderService? = project.serviceIfNotDisposed()
+
+fun getSnykProjectSettingsStateService(project: Project): SnykProjectSettingsStateService? = project.serviceIfNotDisposed()
```
Comment (bastiandoetsch) [inline src/main/kotlin/io/snyk/plugin/Utils.kt:60]: Please break line :)

File: src/main/kotlin/io/snyk/plugin/settings/SnykProjectSettingsConfigurable.kt [hunk 4]
```diff
@@ -4,6 +4,8 @@ import com.intellij.openapi.components.service
 import com.intellij.openapi.options.SearchableConfigurable
```
Comment (bastiandoetsch) [inline src/main/kotlin/io/snyk/plugin/settings/SnykProjectSettingsConfigurable.kt:4]: Please optimize imports :)

File: src/main/kotlin/io/snyk/plugin/settings/SnykProjectSettingsConfigurable.kt [hunk 5]
```diff
@@ -59,10 +61,10 @@ class SnykProjectSettingsConfigurable(val project: Project) : SearchableConfigur
         snykSettingsDialog.saveScanTypeChanges()
 
         if (isProjectSettingsAvailable(project)) {
-            project.service<SnykProjectSettingsStateService>().additionalParameters = snykSettingsDialog.getAdditionalParameters()
+            getSnykProjectSettingsStateService(project)?.additionalParameters = snykSettingsDialog.getAdditionalParameters()
         }
```
Comment (bastiandoetsch) [inline src/main/kotlin/io/snyk/plugin/settings/SnykProjectSettingsConfigurable.kt:65]: Please break line

File: src/main/kotlin/io/snyk/plugin/settings/SnykProjectSettingsConfigurable.kt [hunk 6]
```diff
@@ -86,5 +88,5 @@ class SnykProjectSettingsConfigurable(val project: Project) : SearchableConfigur
 
     private fun isAdditionalParametersModified(): Boolean =
         isProjectSettingsAvailable(project)
-            && snykSettingsDialog.getAdditionalParameters() != project.service<SnykProjectSettingsStateService>().additionalParameters
+            && snykSettingsDialog.getAdditionalParameters() != getSnykProjectSettingsStateService(project)?.additionalParameters
 }
```
Comment (bastiandoetsch) [inline src/main/kotlin/io/snyk/plugin/settings/SnykProjectSettingsConfigurable.kt:92]: Please handle Detekt warnings

File: src/main/kotlin/io/snyk/plugin/ui/toolwindow/SnykToolWindowPanel.kt [hunk 7]
```diff
@@ -31,6 +31,8 @@ import io.snyk.plugin.events.SnykResultsFilteringListener
 import io.snyk.plugin.events.SnykScanListener
 import io.snyk.plugin.events.SnykSettingsListener
 import io.snyk.plugin.events.SnykTaskQueueListener
+import io.snyk.plugin.getAmplitudeExperimentService
```
Comment (bastiandoetsch) [inline src/main/kotlin/io/snyk/plugin/ui/toolwindow/SnykToolWindowPanel.kt:34]: Looks like the imports need some organizing

File: src/integTest/kotlin/io/snyk/plugin/services/SnykTaskQueueServiceHeavyTest.kt (hunk 1 above)
Comment (ArtsiomCh) [reply src/integTest/kotlin/io/snyk/plugin/services/SnykTaskQueueServiceHeavyTest.kt:17]: afaik no errors/exceptions from tests goes to Sentry, as the key (?) is deployed only into prod versions of the plugin

**Review by bastiandoetsch — APPROVED** [review]

**Pull request merged**